components/automate-ui/src/app/pages/job-add/job-add.validators.ts

```diff
@@ -1,19 +1,19 @@
 import {
+  AbstractControl,
   FormArray,
-  FormGroup,
   ValidationErrors,
   ValidatorFn
 } from '@angular/forms';
 
 export const nodeSelectionRequiredValidator: ValidatorFn =
-  (control: FormGroup): ValidationErrors | null => {
+  (control: AbstractControl): ValidationErrors | null => {
```

Comment:
Abstract control is the replacement for `FormGroup` in this case as a result of the FormBuilder#group deprecation.
ref: https://angular.io/api/forms/FormBuilder#group

Reply:
I am not seeing any mention of `FormGroup` being deprecated. 😕

Reply:
From about halfway down on https://angular.io/api/forms/FormBuilder#group

>Construct a new FormGroup instance.
>
>~~`group(controlsConfig: { [key: string]: any; }, options: { [key: string]: any; }): FormGroup`~~
>
>**Deprecated** This API is not typesafe and can result in issues with Closure Compiler renaming. Use the FormBuilder#group overload with AbstractControlOptions instead. Note that AbstractControlOptions expects validators and asyncValidators to be valid validators. If you have custom validators, make sure their validation function parameter is AbstractControl and not a sub-class, such as FormGroup. These functions will be called with an object of type AbstractControl and that cannot be automatically downcast to a subclass, so TypeScript sees this as an error. For example, change the (group: FormGroup) => ValidationErrors|null signature to be (group: AbstractControl) => ValidationErrors|null.

The main portion for our concerns here are these two sentences:
>If you have custom validators, make sure their validation function parameter is AbstractControl and not a sub-class, such as FormGroup. 

>change the (group: FormGroup) => ValidationErrors|null signature to be (group: AbstractControl) => ValidationErrors|null.

Since these are custom validators, they needed to be updated with `AbstractControl`.  I must be mistaken in saying that FormBuilder#group is deprecated - the paragraph above will have to stand in here as a more accurate definition.